File: src/main/java/destiny/astrology/classical/EssentialFaceDefaultImplTest.java
/**
 * @author smallufo 
 * Created on 2007/11/28 at 下午 9:20:46
 */ 
package destiny.astrology.classical;

import junit.framework.TestCase;
import destiny.astrology.Planet;
import destiny.astrology.ZodiacSign;

public class EssentialFaceDefaultImplTest extends TestCase
{
  public void testGetStarDouble()
  {
    EssentialFaceIF impl = new EssentialFaceDefaultImpl();
    
    assertTrue(impl.getFaceStar(0)==Planet.MARS);
    assertTrue(impl.getFaceStar(9.99)==Planet.MARS);
    assertTrue(impl.getFaceStar(10)==Planet.SUN);
    assertTrue(impl.getFaceStar(19.99)==Planet.SUN);
    assertTrue(impl.getFaceStar(20)==Planet.VENUS);
    assertTrue(impl.getFaceStar(20.99)==Planet.VENUS);
    
    assertTrue(impl.getFaceStar(30)==Planet.MERCURY);
    assertTrue(impl.getFaceStar(40)==Planet.MOON);
    assertTrue(impl.getFaceStar(50)==Planet.SATURN);
    
    assertTrue(impl.getFaceStar(60)==Planet.JUPITER);
    assertTrue(impl.getFaceStar(70)==Planet.MARS);
    assertTrue(impl.getFaceStar(80)==Planet.SUN);
    
    assertTrue(impl.getFaceStar( 90)==Planet.VENUS);
    assertTrue(impl.getFaceStar(100)==Planet.MERCURY);
    assertTrue(impl.getFaceStar(110)==Planet.MOON);
    
    assertTrue(impl.getFaceStar(120)==Planet.SATURN);
    assertTrue(impl.getFaceStar(130)==Planet.JUPITER);
    assertTrue(impl.getFaceStar(140)==Planet.MARS);
    
    assertTrue(impl.getFaceStar(150)==Planet.SUN);
    assertTrue(impl.getFaceStar(160)==Planet.VENUS);
    assertTrue(impl.getFaceStar(170)==Planet.MERCURY);
    
    assertTrue(impl.getFaceStar(180)==Planet.MOON);
    assertTrue(impl.getFaceStar(190)==Planet.SATURN);
    assertTrue(impl.getFaceStar(200)==Planet.JUPITER);
    
    assertTrue(impl.getFaceStar(210)==Planet.MARS);
    assertTrue(impl.getFaceStar(220)==Planet.SUN);
    assertTrue(impl.getFaceStar(230)==Planet.VENUS);
    
    assertTrue(impl.getFaceStar(240)==Planet.MERCURY);
    assertTrue(impl.getFaceStar(250)==Planet.MOON);
    assertTrue(impl.getFaceStar(260)==Planet.SATURN);
    
    assertTrue(impl.getFaceStar(270)==Planet.JUPITER);
    assertTrue(impl.getFaceStar(280)==Planet.MARS);
    assertTrue(impl.getFaceStar(290)==Planet.SUN);
    
    assertTrue(impl.getFaceStar(300)==Planet.VENUS);
    assertTrue(impl.getFaceStar(310)==Planet.MERCURY);
    assertTrue(impl.getFaceStar(320)==Planet.MOON);
    
    assertTrue(impl.getFaceStar(330)==Planet.SATURN);
    assertTrue(impl.getFaceStar(340)==Planet.JUPITER);
    assertTrue(impl.getFaceStar(350)==Planet.MARS);
    assertTrue(impl.getFaceStar(359.99)==Planet.MARS);
    //破360
    assertTrue(impl.getFaceStar(360)==Planet.MARS);
    assertTrue(impl.getFaceStar(361)==Planet.MARS);
    assertTrue(impl.getFaceStar(369.99)==Planet.MARS);
    assertTrue(impl.getFaceStar(720)==Planet.MARS);
    assertTrue(impl.getFaceStar(721)==Planet.MARS);
    assertTrue(impl.getFaceStar(729.99)==Planet.MARS);
    
  }

  public void testGetStarZodiacSignDouble()
  {
    EssentialFaceIF impl = new EssentialFaceDefaultImpl();

    //戌
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 0) == Planet.MARS);
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 9.99) == Planet.MARS);
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 10) == Planet.SUN);
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 19.99) == Planet.SUN);
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 20) == Planet.VENUS);
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 29.99) == Planet.VENUS);
    //破30
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 30) == Planet.MERCURY);
    assertTrue(impl.getFaceStar(ZodiacSign.ARIES, 39.99) == Planet.MERCURY);
    
    //亥
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 0) == Planet.SATURN);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 9.99) == Planet.SATURN);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 10) == Planet.JUPITER);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 19.99) == Planet.JUPITER);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 20) == Planet.MARS);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 29.99) == Planet.MARS);
    //破30 , 回到戌
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 30) == Planet.MARS);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 39.990) == Planet.MARS);
    assertTrue(impl.getFaceStar(ZodiacSign.PISCES, 40) == Planet.SUN);
  }

}

File: src/test/java/destiny/astrology/classical/EssentialFaceDefaultImplTest.kt
/**
 * @author smallufo
 * Created on 2007/11/28 at 下午 9:20:46
 */
package destiny.astrology.classical

import destiny.astrology.Planet
import destiny.astrology.ZodiacSign
import kotlin.test.Test
import kotlin.test.assertSame

class EssentialFaceDefaultImplTest {
  @Test
  fun testGetStarDouble() {
    val impl = EssentialFaceDefaultImpl()

    assertSame(impl.getFaceStar(0.0) , Planet.MARS)
    assertSame(impl.getFaceStar(9.99) , Planet.MARS)
    assertSame(impl.getFaceStar(10.0) , Planet.SUN)
    assertSame(impl.getFaceStar(19.99) , Planet.SUN)
    assertSame(impl.getFaceStar(20.0) , Planet.VENUS)
    assertSame(impl.getFaceStar(20.99) , Planet.VENUS)

    assertSame(impl.getFaceStar(30.0) , Planet.MERCURY)
    assertSame(impl.getFaceStar(40.0) , Planet.MOON)
    assertSame(impl.getFaceStar(50.0) , Planet.SATURN)

    assertSame(impl.getFaceStar(60.0) , Planet.JUPITER)
    assertSame(impl.getFaceStar(70.0) , Planet.MARS)
    assertSame(impl.getFaceStar(80.0) , Planet.SUN)

    assertSame(impl.getFaceStar(90.0) , Planet.VENUS)
    assertSame(impl.getFaceStar(100.0) , Planet.MERCURY)
    assertSame(impl.getFaceStar(110.0) , Planet.MOON)

    assertSame(impl.getFaceStar(120.0) , Planet.SATURN)
    assertSame(impl.getFaceStar(130.0) , Planet.JUPITER)
    assertSame(impl.getFaceStar(140.0) , Planet.MARS)

    assertSame(impl.getFaceStar(150.0) , Planet.SUN)
    assertSame(impl.getFaceStar(160.0) , Planet.VENUS)
    assertSame(impl.getFaceStar(170.0) , Planet.MERCURY)

    assertSame(impl.getFaceStar(180.0) , Planet.MOON)
    assertSame(impl.getFaceStar(190.0) , Planet.SATURN)
    assertSame(impl.getFaceStar(200.0) , Planet.JUPITER)

    assertSame(impl.getFaceStar(210.0) , Planet.MARS)
    assertSame(impl.getFaceStar(220.0) , Planet.SUN)
    assertSame(impl.getFaceStar(230.0) , Planet.VENUS)

    assertSame(impl.getFaceStar(240.0) , Planet.MERCURY)
    assertSame(impl.getFaceStar(250.0) , Planet.MOON)
    assertSame(impl.getFaceStar(260.0) , Planet.SATURN)

    assertSame(impl.getFaceStar(270.0) , Planet.JUPITER)
    assertSame(impl.getFaceStar(280.0) , Planet.MARS)
    assertSame(impl.getFaceStar(290.0) , Planet.SUN)

    assertSame(impl.getFaceStar(300.0) , Planet.VENUS)
    assertSame(impl.getFaceStar(310.0) , Planet.MERCURY)
    assertSame(impl.getFaceStar(320.0) , Planet.MOON)

    assertSame(impl.getFaceStar(330.0) , Planet.SATURN)
    assertSame(impl.getFaceStar(340.0) , Planet.JUPITER)
    assertSame(impl.getFaceStar(350.0) , Planet.MARS)
    assertSame(impl.getFaceStar(359.99) , Planet.MARS)
    //破360
    assertSame(impl.getFaceStar(360.0) , Planet.MARS)
    assertSame(impl.getFaceStar(361.0) , Planet.MARS)
    assertSame(impl.getFaceStar(369.99) , Planet.MARS)
    assertSame(impl.getFaceStar(720.0) , Planet.MARS)
    assertSame(impl.getFaceStar(721.0) , Planet.MARS)
    assertSame(impl.getFaceStar(729.99) , Planet.MARS)

  }

  @Test
  fun testGetStarZodiacSignDouble() {
    val impl = EssentialFaceDefaultImpl()

    //戌
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 0.0) , Planet.MARS)
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 9.99) , Planet.MARS)
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 10.0) , Planet.SUN)
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 19.99) , Planet.SUN)
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 20.0) , Planet.VENUS)
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 29.99) , Planet.VENUS)
    //破30
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 30.0) , Planet.MERCURY)
    assertSame(impl.getFaceStar(ZodiacSign.ARIES, 39.99) , Planet.MERCURY)

    //亥
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 0.0) , Planet.SATURN)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 9.99) , Planet.SATURN)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 10.0) , Planet.JUPITER)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 19.99) , Planet.JUPITER)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 20.0) , Planet.MARS)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 29.99) , Planet.MARS)
    //破30 , 回到戌
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 30.0) , Planet.MARS)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 39.990) , Planet.MARS)
    assertSame(impl.getFaceStar(ZodiacSign.PISCES, 40.0) , Planet.SUN)
  }

}
